modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/lifecycle/EventRemotePropagatorExportImportLifecycleListener.java

```diff
@@ -138,48 +144,38 @@ private boolean _eventNeedsToBePropagated(
 			return false;
 		}
 
-		String remoteAddress = GetterUtil.getString(
-			settingsMapOptional.map(
-				settingsMap -> settingsMap.get("remoteAddress")
-			).orElse(
-				StringPool.BLANK
-			));
+		if (remoteAddressValue != null) {
+			String remoteAddress = GetterUtil.getString(remoteAddressValue);
+
+			return !remoteAddress.equals("localhost");
+		}
 
-		return !remoteAddress.equals("localhost");
+		return true;
 	}
 
-	private Optional<ExportImportConfiguration> _getExportImportConfiguration(
+	private ExportImportConfiguration _getExportImportConfiguration(
 		ExportImportLifecycleEvent exportImportLifecycleEvent) {
 
 		List<Serializable> attributes =
 			exportImportLifecycleEvent.getAttributes();
 
-		return Optional.ofNullable(
-			(ExportImportConfiguration)attributes.get(0));
+		return (ExportImportConfiguration)attributes.get(0);
 	}
 
-	private Optional<HttpPrincipal> _getHttpPrincipal(
+	private HttpPrincipal _getHttpPrincipal(
 		ExportImportLifecycleEvent exportImportLifecycleEvent) {
 
-		Optional<ExportImportConfiguration> exportImportConfigurationOptional =
+		ExportImportConfiguration exportImportConfiguration =
 			_getExportImportConfiguration(exportImportLifecycleEvent);
 
-		return exportImportConfigurationOptional.map(
-			exportImportConfiguration ->
-				exportImportConfiguration.getSettingsMap()
-		).map(
-			settingsMap -> MapUtil.getLong(settingsMap, "userId")
-		).map(
-			userId -> _userLocalService.fetchUser(userId)
-		).flatMap(
-			user -> _getHttpPrincipal(
-				user, _getRemoteURL(exportImportLifecycleEvent))
-		);
+		return _getHttpPrincipal(
```

Comment:
@JanisZhang Please also fix this one.

Basically, each `.map` (or maybe `.flatMap` also) should be converted into if-null-return-null. For example, what if the `settingsMap` doesn't contain `userId`? The local service will query the user with `0`, which may return null.

If user is null, the original `_getHttpPrincipal` method will return `Optional.empty()`, while the changed version will throw NPE.

cc @W2jx @slnn @yuhai @LisztLiszt